### PR TITLE
refactor: use absolute include paths

### PR DESCRIPTION
**Pull request by therealbobo**
**What type of PR is this?**

/kind cleanup
/kind design
/kind feature

**Any specific area of the project related to this PR?**

/area libscap
/area libpman
/area libsinsp
/area tests

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

This PR proposes to enhance the maintainability of the `libs` components by updating the include paths for `libsinsp`, `libscap`, and `driver`. The motivation behind this proposal is to improve readability of the code and provide a more explicit and organized structure for including header files.

The proposed patch brings:

- **Clarity of Ownership**:
Using a more verbose include path ensures clear identification of the owner of each header file. This helps in understanding the origin and purpose of included headers at a glance avoids any possible ambiguity (e.g. `utils.h` from libsinsp or `utils.h` from libscap)

- **Streamlined CMake Configuration**:
The proposed include paths will align with a more streamlined CMake configuration. Only a minimal set of directory will be included: the LIBS_DIR (`/userspace`), the driver directory (`/driver`), and some dynamically generated directories (such as `${PROJECT_BINARY_DIRECTORY}/driver/src/` where the generated `driver_config.h` is placed). This minimizes the risk of including unnecessary or conflicting headers. Furthermore, given that the proposed change will touch the cmake setup, probably could be worth adopt a more modern approach replacing the global `include_directories` with the more granular `target_include_directories` approach.

Specifically, I propose transitioning from the current convention:
```
#include "strl.h"
#include "sinsp.h"
#include "sinsp_int.h"
#include "chisel.h"
#include "chisel_api.h"
#include "filter.h"
#include "filterchecks.h"
```
to a more explicit format:
```
#include <libscap/strl.h>
#include <libsinsp/sinsp.h>
#include <libsinsp/sinsp_int.h>
#include <chisel/chisel.h>
#include <chisel/chisel_api.h>
#include <libsinsp/filter.h>
#include <libsinsp/filterchecks.h>
```
    
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

I understand this can be disruptive for some of the adopters of the Falco libs, so this PR acts as a POC and is totally open to feedback and opinions. Overall, the drive for this is to make the libs a more long-sustainable and high-quality codebase.

To illustrate the viability and benefits of the proposed change, I have implemented a patch in my branch of falco. Please find the link to the branch below:

- https://github.com/therealbobo/falco/tree/cleanup-relative-import-paths

**Does this PR introduce a user-facing change?**:

```release-note
libs consumers have to change the include paths.
```

### CODE REVIEW

**Comment by jasondellaluce** [comment]
This is amazing @therealbobo . +1 on my end, this is quite a big leap toward making the libs true maintainable libraries.

File: userspace/libscap/CMakeLists.txt [hunk 1]
```diff
@@ -32,36 +31,33 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
 	endif()
 
 	# do not remove this since when WITH_DRIVER is off
-	if(NOT DEFINED DRIVER_NAME)
-		set(DRIVER_NAME "scap")
-	endif()
+	set(DRIVER_NAME "scap" CACHE STRING _)
```
Comment (Molter73) [inline userspace/libscap/CMakeLists.txt:34]: Why is this change needed? If an adopter of the libraries wants to change the name for the driver they should be allowed to, don't you think?

File: userspace/libscap/CMakeLists.txt [hunk 2]
```diff
@@ -32,36 +31,33 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
...
 
 	string(REPLACE "-" "_" SCAP_KERNEL_MODULE_NAME "${DRIVER_NAME}")
 	add_definitions(-DSCAP_KERNEL_MODULE_NAME="${SCAP_KERNEL_MODULE_NAME}")
 endif()
 
-if(NOT DEFINED SCAP_HOST_ROOT_ENV_VAR_NAME)
-	set(SCAP_HOST_ROOT_ENV_VAR_NAME "HOST_ROOT")
-endif()
+set(SCAP_HOST_ROOT_ENV_VAR_NAME "HOST_ROOT" CACHE STRING _)
```
Comment (Molter73) [inline userspace/libscap/CMakeLists.txt:40]: Similar question here, why is `SCAP_HOST_ROOT_ENV_VAR_NAME` hardcoded now?

File: userspace/libscap/CMakeLists.txt [hunk 3]
```diff
@@ -32,36 +31,33 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
...
 add_definitions(-DSCAP_HOST_ROOT_ENV_VAR_NAME="${SCAP_HOST_ROOT_ENV_VAR_NAME}")
 
-if(NOT DEFINED SCAP_HOSTNAME_ENV_VAR)
-	set(SCAP_HOSTNAME_ENV_VAR "SCAP_HOSTNAME")
-endif()
+set(SCAP_HOSTNAME_ENV_VAR "SCAP_HOSTNAME" CACHE STRING _)
```
Comment (Molter73) [inline userspace/libscap/CMakeLists.txt:43]: Same question here.

**Review by Molter73 — COMMENTED** [review]
Hey folks! Unfortunately I'm on leave and don't have the time to do a full review of the PR. Skimming through it, it looks good, just have a couple questions on a few changes that seem unnecessary from my perspective that could also make it harder on adopters to customize things.

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by incertum — APPROVED** [review]
/approve

**Pull request merged**